### PR TITLE
feat(desktop/chat) Updating chat/communities/categories due to backend changes

### LINKED ISSUE #4062: refactor: chat section changes needed for backend refactor

### PR DESCRIPTION
**Pull request by alexandraB99**
Closes #4062

### What does the PR do
The models of chat and communities with categories in the backend have changed as well as the current active application. This PR introduces changes in the UI corresponding to the backend changes in order to follow up with these.

### Affected areas
Chat/communities view

### CODE REVIEW

File: ui/app/AppLayouts/Chat/views/CommunityColumnView.qml [hunk 1]
```diff
@@ -129,10 +128,8 @@ Item {
               
             draggableItems: root.store.chatsModelInst.communities.activeCommunity.admin
             draggableCategories: root.store.chatsModelInst.communities.activeCommunity.admin
-            chatList.model: root.store.chatsModelInst.communities.activeCommunity.chats
-
-            categoryList.model: root.store.chatsModelInst.communities.activeCommunity.categories
-
+            chatList.model: store.model
+            categoryList.model: store.model.subItems
```
Comment (PascalPrecht) [inline ui/app/AppLayouts/Chat/views/CommunityColumnView.qml:132]: I know this is not scope for this PR, but I just want to leave this here:

The fact that the new model is called `model.subItems` makes it very hard to know what data one is dealing with here. I think it'd be better if we made model expressions more descriptive.

File: ui/app/AppMain.qml [hunk 2]
```diff
@@ -352,44 +352,42 @@ Item {
...
-                else if(mainModule.activeSection.sectionType === Constants.appSection.profile) {
-                    return Constants.appViewStackIndex.profile
+                    return 5;
                 }
                 else if(mainModule.activeSection.sectionType === Constants.appSection.node) {
-                    return Constants.appViewStackIndex.node
+                    return 6;
+                }
+                else if(mainModule.activeSection.sectionType === Constants.appSection.profile) {
+                    return 7;
```
Comment (PascalPrecht) [inline ui/app/AppMain.qml:389]: I'd appreciate if we'd re-introduce an equivalent to `Constants.appViewStackIndex.node` etc

**Review by PascalPrecht — COMMENTED** [review]
I **think** the changes look good apart from the comments I've left.

It'd be create if the commit message had a bit more useful information on what these changes are doing. Very hard for me to tell if everything is as intended because the intention isn't clear...

Also judging by the commit, it seems this is a work-in-progress. Is that intended?

**Comment by PascalPrecht** [comment]
@saledjenic this is ready for another review

File: ui/app/AppMain.qml (hunk 2 above)
Comment (alexandraB99) [reply ui/app/AppMain.qml]: > 

done

File: ui/app/AppLayouts/Chat/views/CommunityColumnView.qml (hunk 1 above)
Comment (alexandraB99) [reply ui/app/AppLayouts/Chat/views/CommunityColumnView.qml:132]: yes, this will change with the latest updates in [StatusQ](https://github.com/status-im/StatusQ/pull/486)

File: ui/imports/utils/Constants.qml [hunk 3]
```diff
@@ -20,17 +20,6 @@ QtObject {
         readonly property int node: 7
     }
 
-    readonly property QtObject appViewStackIndex: QtObject {
```
Comment (saledjenic) [inline ui/imports/utils/Constants.qml:23]: @alexandraB99 why did you remove this? This is very important and mandatory part in order to have everything in place correctly.

File: ui/app/AppMain.qml [hunk 4]
```diff
@@ -350,76 +350,49 @@ Item {
         appView: StackLayout {
             id: appView
             anchors.fill: parent
-            currentIndex: {
```
Comment (saledjenic) [inline ui/app/AppMain.qml:353]: @alexandraB99 why did you remove this? Please analyze again this code. Actually changes needed due to changed model, which is used for displaying chats/categories/channels, do not have anything and don't affect anyhow this part here.

File: ui/app/AppMain.qml [hunk 5]
```diff
@@ -350,76 +350,49 @@ Item {
...
-                }
-                else if(mainModule.activeSection.sectionType === Constants.appSection.node) {
-                    return Constants.appViewStackIndex.node
-                }
-
-                // We should never end up here
-                console.error("Unknown section type")
-            }
-
+            currentIndex: appMain.rootStore.appViewIndex
```
Comment (saledjenic) [inline ui/app/AppMain.qml:353]: @alexandraB99 `appMain.rootStore.appViewIndex` you used here refers to the section type, not to the index of a component in this `StackLayout`.

File: ui/app/AppMain.qml [hunk 6]
```diff
@@ -350,76 +350,49 @@ Item {
...
             onCurrentIndexChanged: {
-                var obj = this.children[currentIndex];
-                if(!obj)
-                    return
+                commLoader.active = false;
+                if (currentIndex === Constants.appSection.chat) {
```
Comment (saledjenic) [inline ui/app/AppMain.qml:356]: @alexandraB99 as I see you accepted changes from the master branch here, but they are not applicable in the context of refactored code we have in `base_bc`

File: ui/app/AppMain.qml [hunk 7]
```diff
@@ -498,45 +495,11 @@ Item {
...
 
-            WalletV2Layout {
-                id: walletV2LayoutContainer
+            UIComponents {
                 Layout.fillWidth: true
                 Layout.alignment: Qt.AlignLeft | Qt.AlignTop
                 Layout.fillHeight: true
             }
-
-            Repeater{
```
Comment (saledjenic) [inline ui/app/AppMain.qml:508]: @alexandraB99 why is this deleted? This must be here and it must be the last children of `StackLayout` component. Without this we won't have `ChatLayout` for each community.

File: ui/main.qml [hunk 8]
```diff
@@ -69,7 +69,7 @@ StatusWindow {
 
     Action {
         shortcut: "Ctrl+W"
-        enabled: loader.item && !!loader.item.appLayout? loader.item.appLayout.appView.currentIndex === Constants.appViewStackIndex.browser
+        enabled: loader.item && !!loader.item.appLayout? loader.item.appLayout.appView.currentIndex === Constants.appSection.browser
```
Comment (saledjenic) [inline ui/main.qml:72]: @alexandraB99 as mentioned above we cannot mix section types and indexes of a sections as component added to the `StackLayout`.

**Review by saledjenic — CHANGES_REQUESTED** [review]
@alexandraB99 I have many questions about this change, and some of them I added commenting the changes applied here.

But in front of all the main question I have is actually if your `StatusQ` related changes for handling new model related to chat/categories/channels are done here: https://github.com/status-im/StatusQ/pull/486 why that is not checked out in `ui/StatusQ`here?

And one thing more, since `base_bc` branch for `status-desktop` relies on `StatusQ` changes done in the `base_bc` branch in there (PR for that is [here](https://github.com/status-im/StatusQ/pull/489)) then your `StatusQ` changes should be on top of that branch.

**Comment by saledjenic** [comment]
@alexandraB99 I am suggesting to revert these changes and checkout changes you did here: https://github.com/status-im/StatusQ/pull/486 (I added some comments more like improvements there, but in general your changes there are good and work well), and then do here only changes related to the changed model which should be used for displaying chat/categories/channels. It has exactly the same exposed items and properties as a fake model you used in StatusQ.

Here is all about model you should use here:
- [model](https://github.com/status-im/status-desktop/blob/c67384367570031b67bea1bcc24c30966d99e05f/src/app/modules/main/chat_section/model.nim)
- [submodel](https://github.com/status-im/status-desktop/blob/c67384367570031b67bea1bcc24c30966d99e05f/src/app/modules/main/chat_section/sub_model.nim)
- [exposed view and related properties](https://github.com/status-im/status-desktop/blob/c67384367570031b67bea1bcc24c30966d99e05f/src/app/modules/main/chat_section/view.nim)

If you need anything more I can add, but I guess all we need in order to have this in place is already prepared on the BE side, also components are ready in the StatusQ (just need to be rebased on top of `base_bc` branch there) and only remains to merge those two sides here (update qml a bit).

**Comment by alexandraB99** [comment]
@saledjenic I am sorry but you are confusing me a lot with your back and forth requests, I do not end up understanding what exactly you need me to do since it seems that you have everything under control. What if I only work on StatusQ changes and you take it from there? I will close this PR since it seems it doesn't make any sense and I'll work on the changes you asked in [StatusQ](https://github.com/status-im/StatusQ/pull/486) and let you know how to use it (extremely simple) which is what I proposed to you 10days ago where I was again confused with what exactly needs to be done here. @PascalPrecht as we already discussed, could we create the base_bc branch in StatusQ?

**Comment by PascalPrecht** [comment]
@alexandraB99 it seems @saledjenic has already created such a branch here: https://github.com/status-im/StatusQ/pull/489

**Comment by alexandraB99** [comment]
> ~~@saledjenic I am sorry but you are confusing me a lot with your back and forth requests, I do not end up understanding what exactly you need me to do since it seems that you have everything under control. What if I only work on StatusQ changes and you take it from there? I will close this PR since it seems it doesn't make any sense and I'll work on the changes you asked in [StatusQ](https://github.com/status-im/StatusQ/pull/486) and let you know how to use it (extremely simple) which is what I proposed to you 10days ago where I was again confused with what exactly needs to be done here. @PascalPrecht as we already discussed, could we create the base_bc branch in StatusQ?~~

Maybe let me better rephrase.. @saledjenic this commit improves a bit the previous code. Instead of using a repeater is uses a Loader and loads the communities on demand, same for chat. However there is an issue with channels and categories of communities where appear to be the same in all communities, I am currently checking whether this is a StatusQ or backend problem. As about the stack index, since now this is happening in mainModule, it seems we don't need it. Have you tested the PR before commenting? As on my side it works, I would really like you to tell me what exactly doesn't work. It's not clear if the intention here is to just fix/update what you're asking or to also propose improvements on UI side. If it's the first one, then I would just close this PR and concentrate only on the StatusQ changes and supporting you on using the API in desktop. Please let me know.

**Comment by saledjenic** [comment]
@alexandraB99 sorry, my intention was not to confuse you at all, but I did the best I could to explain and help somehow with suggestion what need to be done, to make it easier for you. Your changes in `StatusQ` are fine (the only needed there is to improve some things following comments I added, as I see you already did some of them), also backend part seems good, we need only a changes here, and now I see that you updated the code and add loaders but, still remains the following:

- `chatLoader` is ok since we have a single chat
- `communityLoader` is not cause it is added only once, it must be in a repeater since we need to have as many component as many communities we have, another problematic thing here is that community component/s must be the last component/s added to the stack layout since we don't know in advance how many communities we have, that the thing resolved in the runtime of the app
- the thing I was writing above about mixing section types and indexes is still there, so that must be reverted

@alexandraB99 all in all I will try now to apply statusq changes you did to the desktop app (but in a new branch).

**Comment by saledjenic** [comment]
@alexandraB99 I am closing this PR, since we have everything in place now here:
- https://github.com/status-im/status-desktop/pull/4196
